### ISSUE TITLE
Error messages should include rule name

### ISSUE THREAD

**Issue by mgol** [palantir/tslint#3290]
### Bug Report

- __TSLint version__: 5.7.0
- __TypeScript version__: 2.3.4
- __Running TSLint via__: CLI

#### TypeScript code being linted

```ts
Promise.resolve('foo');
```

with `tslint.json` configuration:

```json
{
  "rules": {
    "no-floating-promises": true
  }
}
```

#### Actual behavior
Running `tslint --project . file.ts` reports:
```
ERROR: /full/path/to/files.ts[1, 1]: Promises must be handled appropriately
```

#### Expected behavior
I'd expect something like:
```
ERROR (no-floating-promises): /full/path/to/files.ts[1, 1]: Promises must be handled appropriately
```

Sometimes an error might be unexpected and we might want to disable the rule that causes it. When you have a large config, though, it may not be obvious which rule reports a particular error.

ESLint formatters print the name of the rule that detects the error.

**Comment by ajafff** [comment]
Use the `verbose` formatter by adding `-t verbose` CLI command.

The rest of the issue is a duplicate of #2228
